### PR TITLE
doc: add a release procedures section

### PR DESCRIPTION
**Pull request by tkatila**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -276,8 +280,19 @@ matching Kubernetes versions are listed below:
 | release-0.15      | Kubernetes 1.15 branch v1.15.x | unsupported |
 | release-0.11      | Kubernetes 1.11 branch v1.11.x | unsupported |
 
-[Go environment]: https://golang.org/doc/install
-[Kubernetes cluster]: https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/
+*Note:* Device plugins leverage the Kubernetes v1 API. The API itself is GA (generally available) and doesn't change between Kubernetes versions. One doesn't necessarily need use the latest Kubernetes cluster with the latest device plugin version. Using a newer device plugins release should work without issues on an older Kubernetes cluster. One possible exception to this are the device plugins CRDs that can vary between versions.
```
Comment (eero-t) [inline README.md:283]: Typo fix (+ more formal spelling):
```suggestion
*Note:* Device plugins leverage the Kubernetes v1 API. The API itself is GA (generally available) and does not change between Kubernetes versions. One does not necessarily need to use the latest Kubernetes cluster with the latest device plugin version. Using a newer device plugins release should work without issues on an older Kubernetes cluster. One possible exception to this are the device plugins CRDs that can vary between versions.
```

File: README.md [hunk 2]
```diff
@@ -276,8 +280,19 @@ matching Kubernetes versions are listed below:
...
+
+Project's release cadence is tied to Kubernetes release cadence. Device plugins release typically follows a couple of weeks after the Kubernetes release. There can be some delays on the releases due to required changes in the pull request pipeline. Once the content is available in the `main` branch and CI & e2e validation <span style="color:green">PASS</span>es, release branch will be created (e.g. release-0.26). The HEAD of release branch will also be tagged with the corresponding [tag](https://github.com/intel/intel-device-plugins-for-kubernetes/tags) (e.g. v0.26.0).
+
+During the [release creation](https://github.com/intel/intel-device-plugins-for-kubernetes/issues/1393), the project's documentation, deployment files etc. will be [changed](https://github.com/intel/intel-device-plugins-for-kubernetes/pull/1441) to point to the newly created version.
+
+Pull requests that are required for specific release are labeled with the corresponding [milestone](https://github.com/intel/intel-device-plugins-for-kubernetes/milestones).
+
+Patch releases (e.g. 0.26.3) are done on an adhoc basis if there are security issues or minor fixes requested for specific version.
```
Comment (eero-t) [inline README.md:295]: Typo fixes:
```suggestion
Patch releases (e.g. 0.26.3) are done on an ad-hoc basis if there are security issues or minor fixes are requested for a specific version.
```

**Review by eero-t — COMMENTED** [review]
Couple of typos, but otherwise looks good!

File: README.md (hunk 2 above)
Comment (mythi) [reply README.md:295]: we should probably say 'on a need basis' rather than 'ad-hoc'

File: README.md (hunk 1 above)
Comment (mythi) [reply README.md:283]: let's add a link for reference https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-changes

File: README.md [hunk 3]
```diff
@@ -276,8 +280,19 @@ matching Kubernetes versions are listed below:
...
+
+Project's release cadence is tied to Kubernetes release cadence. Device plugins release typically follows a couple of weeks after the Kubernetes release. There can be some delays on the releases due to required changes in the pull request pipeline. Once the content is available in the `main` branch and CI & e2e validation <span style="color:green">PASS</span>es, release branch will be created (e.g. release-0.26). The HEAD of release branch will also be tagged with the corresponding [tag](https://github.com/intel/intel-device-plugins-for-kubernetes/tags) (e.g. v0.26.0).
+
+During the [release creation](https://github.com/intel/intel-device-plugins-for-kubernetes/issues/1393), the project's documentation, deployment files etc. will be [changed](https://github.com/intel/intel-device-plugins-for-kubernetes/pull/1441) to point to the newly created version.
+
+Pull requests that are required for specific release are labeled with the corresponding [milestone](https://github.com/intel/intel-device-plugins-for-kubernetes/milestones).
+
+Patch releases (e.g. 0.26.3) are done on a need basis if there are security issues or minor fixes requested for specific version.
```
Comment (eero-t) [inline README.md:295]:
```suggestion
Patch releases (e.g. 0.26.3) are done on a need basis if there are security issues or minor fixes requested for a specific version.
```

**Review by eero-t — COMMENTED** [review]
/lgtm

File: README.md [hunk 4]
```diff
@@ -276,8 +280,19 @@ matching Kubernetes versions are listed below:
...
+
+### Release procedures
+
+Project's release cadence is tied to Kubernetes release cadence. Device plugins release typically follows a couple of weeks after the Kubernetes release. There can be some delays on the releases due to required changes in the pull request pipeline. Once the content is available in the `main` branch and CI & e2e validation <span style="color:green">PASS</span>es, release branch will be created (e.g. release-0.26). The HEAD of release branch will also be tagged with the corresponding [tag](https://github.com/intel/intel-device-plugins-for-kubernetes/tags) (e.g. v0.26.0).
+
+During the [release creation](https://github.com/intel/intel-device-plugins-for-kubernetes/issues/1393), the project's documentation, deployment files etc. will be [changed](https://github.com/intel/intel-device-plugins-for-kubernetes/pull/1441) to point to the newly created version.
+
+Pull requests that are required for specific release are labeled with the corresponding [milestone](https://github.com/intel/intel-device-plugins-for-kubernetes/milestones).
```
Comment (mythi) [inline README.md:293]: it does not give a comprehensive list so is it worth to mention?

Comment (tkatila) [reply README.md:293]: Yep, correct. I'll remove that sentence.

**Review by mythi — APPROVED** [review]

**Pull request merged**